design/object-bucket/object-bucket.md

```diff
@@ -43,115 +51,159 @@ Add a CephObjectBucketClaim API endpoint and control loop to the Rook-Ceph opera
-1. The CephObjectBucketClaim is garbage collected.
+1. The OBC is garbage collected.
+
+    Note: this does not cause the deletion of the OB.  Since the actual bucket is not deleted, it is reasonable that it should still tracked in Kubernetes.
 
 ---
 
 ## Looking Forward
 
-- Resource Quotas cannot be defined for CRDs.  This limits admin control over how many buckets can be created per object store.  A operator could be made configurable (via ConfigMap) to cap the number of buckets per object store.  A PR exists for enabling quotas on CRDs (https://github.com/kubernetes/kubernetes/pull/72384).  If it is merged into Kubernetes, a ResourceQuota example should be defined for CephObjectBucketClaims.
+- Resource Quotas cannot be defined for CRDs.  This limits admin control over how many buckets can be created per object store.  A operator could be made configurable (via ConfigMap) to cap the number of buckets per object store.  A PR exists for enabling quotas on CRDs (https://github.com/kubernetes/kubernetes/pull/72384).  If it is merged into Kubernetes, a ResourceQuota example should be defined for OBCs.
+
+- Custom Bucket and Object policies are not defined for OBCs.  Currently all user keys will have Object PUT/GET/DELETE and object policy setting permissions.  It would be useful to allow users to link secondary keys with a subset of these permissions to buckets.
 
-- ACL control doesn't exist for CephObjectBucketClaims.  Currently all user end keys will have Object PUT/GET/DELETE.  It would be useful to allow users to request secondary keys with a subset of these ACLs. For instance, an Object GET-only key.
+- A generic S3 user API is not defined in this proposal but should be given serious consideration.  A user API should allow cluster users to generate new S3 users, tenants, and access keys in an object store.  This API could be used as method of manually creating object store users or incorporated into the bucket provisioning automation to generate the user's access key.
 
 ---
 
 ## API Specifications
 
-### Custom Resource Definition
+### OBC Custom Resource Definition
 
 ```yaml
 apiVersion: apiextensions.k8s.io/v1beta1 
 kind: CustomResourceDefinition
 metadata:
-  name: cephobjectbucketclaims.rook.io
+  name: objectbucketclaims.rook.io
 spec:
-  group: ceph.rook.io
+  group: rook.io
   version: v1
   scope: namespaced
   names:
-      kind: CephObjectBucketClaim
-      listKind: CephObjectBucketClaimList
-      singular: cephobjectbucketclaim
-      plural: cephobjectbucketclaims
+      kind: ObjectBucketClaim
+      listKind: ObjectBucketClaimList
+      singular: objectbucketclaim
+      plural: objectbucketclaims
 ```
 
 
-### Custom Resource (User Defined)
+### OBC Custom Resource (User Defined)
 
 ```yaml
-apiVersion: ceph.rook.io/v1
-kind: CephObjectBucketClaim
+apiVersion: rook.io/v1
+kind: ObjectBucketClaim
 metadata:
   name: MY-BUCKET-1 [1]
   namespace: USER-NAMESPACE [2]
```

Comment:
Note that ns is not required and defaults to the namespace where the `kubectl create ...` is executed.